File: HashTableSearch.cpp
#include <iostream>
#include <unordered_map>

using namespace std;

int main() {
    unordered_map<int, string> hashTable;

    // Insert key-value pairs
    hashTable[10] = "Apple";
    hashTable[20] = "Banana";
    hashTable[30] = "Cherry";

    // Search for a key
    int key = 20;
    if (hashTable.find(key) != hashTable.end()) {
        cout << "Element found: " << hashTable[key] << endl;
    } else {
        cout << "Element not found!" << endl;
    }

    // Delete a key
    hashTable.erase(10);

    return 0;
}

File: Python/HashTableSearch.py
class HashTable:
    def __init__(self):
        self.table = {}

    def insert(self, key, value):
        self.table[key] = value

    def search(self, key):
        return self.table.get(key, -1)


# Example usage

hash_table = HashTable()
hash_table.insert(10, "Apple")
hash_table.insert(20, "Banana")
hash_table.insert(30, "Cherry")

key = 20
result = hash_table.search(key)
print("Element found:", result)
